contributors/design-proposals/daemonset-update.md

```diff
@@ -158,9 +173,10 @@ For each pending DaemonSet updates, it will:
 1. Check `DaemonSetUpdateStrategy`:
    - If `OnDelete`: do nothing
    - If `RollingUpdate`:
-     - Compare spec of the daemon pods from step 1 with DaemonSet
-       `.spec.template.spec` to see if DaemonSet spec has changed.
-     - If DaemonSet spec has changed, compare `MaxUnavailable` with DaemonSet
+     - Find all daemon pods that belong to this DaemonSet using label selectors.
+       - Pods with "pod-template-generation" value equal to the DaemonSet's
```

Comment:
Could we use a nil template generation in order to indicate we expect the label to not be set? In this case the DaemonSet does not attempt the comparison until the first time the upgrade is performed.